plaidml/op/op_test.cc

```diff
@@ -887,140 +476,79 @@ TEST(Op, SpatialPadding) {
       TensorLayout::NXC);        // data layout
   auto program = makeProgram("spatial_padding", {X});
   IVLOG(1, program);
-  EXPECT_THAT(program, Eq(R"#(
-#map0 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2 + 1, d3 + 3)>
-#map1 = affine_map<(d0, d1, d2, d3) -> (d0, d1, d2, d3)>
-
-
-module {
-  func @spatial_padding(%arg0: tensor<64x4x32x32xf32> {tile.name = "A"}) -> tensor<64x4x36x38xf32> {
-    %cst = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> f32
-    %0 = tile.contract assign, none, %cst, %arg0 {idxs = ["n", "c", "x0", "x1"], sink = #map0, srcs = [#map1]} : f32, tensor<64x4x32x32xf32> -> tensor<64x4x36x38xf32>
-    return %0 : tensor<64x4x36x38xf32>
-  }
-}
-)#"));
+  runProgram(program);
 }
 
-TEST(Op, Square) {
+TEST_F(OpTest, Square) {
   auto A = Placeholder(DType::FLOAT32, {10}, "A");
   auto program = makeProgram("square", {op::square(A)});
   IVLOG(1, program);
-  EXPECT_THAT(program, Eq(R"#(
-
-module {
-  func @square(%arg0: tensor<10xf32> {tile.name = "A"}) -> tensor<10xf32> {
-    %0 = "eltwise.mul"(%arg0, %arg0) : (tensor<10xf32>, tensor<10xf32>) -> tensor<10xf32>
-    return %0 : tensor<10xf32>
-  }
-}
-)#"));
+  runProgram(program);
 }
 
-TEST(Op, Sum) {
+TEST_F(OpTest, Sum) {
   auto A = Placeholder(DType::FLOAT32, {10, 20}, "A");
   auto program = makeProgram("sum", {op::sum(A)});
   IVLOG(1, program);
-  EXPECT_THAT(program, Eq(R"#(
-#map0 = affine_map<() -> ()>
-#map1 = affine_map<(d0, d1) -> (d0, d1)>
-
-
-module {
-  func @sum(%arg0: tensor<10x20xf32> {tile.name = "A"}) -> f32 {
-    %cst = "eltwise.sconst"() {value = 0.000000e+00 : f64} : () -> f32
-    %0 = tile.contract add, none, %cst, %arg0 {sink = #map0, srcs = [#map1]} : f32, tensor<10x20xf32> -> f32
-    return %0 : f32
-  }
-}
-)#"));
+  runProgram(program);
 }
 
-TEST(Op, Squeeze) {
-  auto A = Placeholder(DType::FLOAT32, {32, 1, 4, 1}, "A");
-  auto X = op::squeeze(  //
+TEST_F(OpTest, Squeeze) {
+  auto A = Placeholder(DType::UINT64, {3, 1, 4, 1}, "A");
+  auto C = op::squeeze(  //
       A,                 // tensor to squeeze
       {1, 3});           // axes to squeeze
-  auto program = makeProgram("squeeze", {X});
+  auto program = makeProgram("squeeze", {C});
   IVLOG(1, program);
-  EXPECT_THAT(program, Eq(R"#(
-module {
-  func @squeeze(%arg0: tensor<32x1x4x1xf32> {tile.name = "A"}) -> tensor<32x4xf32> {
-    %c32 = tile.constant 32
-    %c4 = tile.constant 4
-    %0 = "tile.reshape"(%arg0, %c32, %c4) : (tensor<32x1x4x1xf32>, index, index) -> tensor<32x4xf32>
-    return %0 : tensor<32x4xf32>
-  }
-}
-)#"));
+
+  std::vector<uint64_t> A_input = {0, 1, 2,  3,  //
+                                   4, 5, 6,  7,  //
+                                   8, 9, 10, 11};
+  std::vector<uint64_t> C_output = {0, 1, 2,  3,  //
+                                    4, 5, 6,  7,  //
+                                    8, 9, 10, 11};
```

Comment:
It might be clearer to use literally the same vector for both the input and the output; since these vectors do not include shape information, Squeeze ought to be a no-op. This is currently reflected by A_input and C_output having the same contents, but it might be more obvious if they weren't different variables.